### ISSUE TITLE
Customise batching function for `findUnique()`

### ISSUE THREAD

**Issue by okomarov** [prisma/prisma#17983]
## Problem

We'd like to remove our dataloader boilerplate and rely on `findUnique` only but Prisma's performance is an **order of magnitude worse** (some complex queries complete in 10s vs 1s).

## Suggested solution

I'd like to customise the batch scheduler to fire off every e.g. 50 ms, like in the Dataloader example [here](https://github.com/graphql/dataloader#batch-scheduling):
```js
const myLoader = new DataLoader(myBatchFn, {
  batchScheduleFn: callback => setTimeout(callback, 50),
});
```

## Alternatives

As an alternative, I've been trying to write a dataloadified `<model>.getById` with Prisma client extensions but I am having TypeScript issues and can't really get anywhere.

```ts 
function getDataloaderById<T>(model: T) {
//                                   ↑ 
//                                  How to type this outside of client extensions?
  new Dataloader(async (ids) => {
    const entities = await model.findMany({
      where: {
        id: {in: [...ids], // turn ids into non-readonly
        },
      },
    });
    const entitiesById = new Map(entities.map((item) => [item.id, item]));
    return ids.map((id) => entitiesById.get(id) ?? null);
  });
}
```

## Additional context

Related issues: https://github.com/prisma/prisma/issues/6573, https://github.com/prisma/prisma/issues/16605 and generally https://github.com/prisma/prisma/labels/topic%3A%20batching

**Comment by ludralph** [comment]
Closing this in favour of [6573](https://github.com/prisma/prisma/issues/6573)
